pkg/syncer/metadatasyncer.go

```diff
@@ -947,59 +947,33 @@ func ReloadConfiguration(metadataSyncer *metadataSyncInformer, reconnectToVCFrom
 		metadataSyncer.clusterFlavor != cnstypes.CnsClusterFlavorGuest {
 		// Vanilla ReloadConfiguration
 		if isMultiVCenterFssEnabled {
-			var multivCenterTopologyDeployment bool
-			if len(metadataSyncer.configInfo.Cfg.VirtualCenter) > 1 {
-				multivCenterTopologyDeployment = true
-			}
 			newVcenterConfigs, err := cnsvsphere.GetVirtualCenterConfigs(ctx, cfg)
 			if err != nil {
 				return logger.LogNewErrorf(log, "failed to get VirtualCenterConfigs. err=%v", err)
 			}
 			if newVcenterConfigs != nil {
 				for _, newVCConfig := range newVcenterConfigs {
 					newVCConfig.ReloadVCConfigForNewClient = true
-					var vcenter *cnsvsphere.VirtualCenter
-					vcenter, err = cnsvsphere.GetVirtualCenterInstanceForVCenterConfig(ctx, newVCConfig, false)
-					if err != nil {
-						if err == cnsvsphere.ErrVCAlreadyRegistered {
-							vcenter, err = cnsvsphere.GetVirtualCenterInstanceForVCenterHost(ctx, newVCConfig.Host, false)
-							if err != nil {
-								return logger.LogNewErrorf(log, "failed to get VirtualCenter. err=%v", err)
-							}
-						} else {
-							return logger.LogNewErrorf(log, "failed to get VirtualCenter. err=%v", err)
-						}
-					}
-					vcenter.Config = newVCConfig
 					if metadataSyncer.volumeManagers[newVCConfig.Host] == nil {
-						log.Infof("creating new volumemanager for vCenter: %q", newVCConfig.Host)
-						volumeManager, err := volumes.GetManager(ctx, vcenter, nil,
-							true, true, multivCenterTopologyDeployment, tasksListViewEnabled)
-						if err != nil {
-							return logger.LogNewErrorf(log, "failed to create an instance of volume manager. err=%v", err)
```

Comment:
And also this removal

Reply:
We do not need to make new VC client right upon password is rotated. We have made a change in the VC Connect method to re-read latest config when existing session is expired or timeout and update the VC client pointer right there.

Reply:
Refer to https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/common/cns-lib/vsphere/virtualcenter.go#L309-L332